api/filters/replacement/replacement.go

```diff
@@ -66,18 +66,44 @@ func rejectId(rejects []*types.Selector, nodeId *types.KrmId) bool {
 
 func applyToNode(node *yaml.RNode, value *yaml.RNode, target *types.TargetSelector) error {
 	for _, fp := range target.FieldPaths {
-		t, err := node.Pipe(yaml.Lookup(strings.Split(fp, ".")...))
+		fieldPath := strings.Split(fp, ".")
+		var t *yaml.RNode
+		var err error
+		if target.Options != nil && target.Options.Create {
+			t, err = node.Pipe(yaml.LookupCreate(value.YNode().Kind, fieldPath...))
+		} else {
+			t, err = node.Pipe(yaml.Lookup(fieldPath...))
+		}
 		if err != nil {
 			return err
 		}
 		if t != nil {
-			// TODO (#3492): Use the field options to refine interpretation of the field
-			t.SetYNode(value.YNode())
+			setTargetValue(target.Options, t, value)
 		}
 	}
 	return nil
 }
 
+func setTargetValue(options *types.FieldOptions, t *yaml.RNode, value *yaml.RNode) {
+	if t.YNode().Kind == yaml.ScalarNode && value.YNode().Kind == yaml.ScalarNode &&
+		options != nil && options.Delimiter != "" {
```

Comment:
if `options.Delimiter` is non-empty but the field types aren't suitable, shouldn't we report an error?

Reply:
+1

Reply:
done